validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/GetGraffiti.java

```diff
@@ -0,0 +1,121 @@
+  private final KeyManager keyManager;
+
+  private static final SerializableTypeDefinition<GraffitiResponse> GRAFFITI_TYPE =
+      SerializableTypeDefinition.object(GraffitiResponse.class)
+          .withOptionalField("pubkey", STRING_TYPE, GraffitiResponse::getPublicKey)
+          .withField("graffiti", STRING_TYPE, GraffitiResponse::getGraffiti)
+          .build();
+
+  private static final SerializableTypeDefinition<GraffitiResponse> RESPONSE_TYPE =
+      SerializableTypeDefinition.object(GraffitiResponse.class)
+          .name("GraffitiResponse")
+          .withField("data", GRAFFITI_TYPE, Function.identity())
+          .build();
+
+  public GetGraffiti(final KeyManager keyManager) {
+    super(
+        EndpointMetadata.get(ROUTE)
+            .operationId("getGraffiti")
+            .summary("Get Graffiti")
+            .description(
+                "Get the graffiti for an individual validator. If no graffiti is set explicitly, returns the process-wide default.")
+            .tags(TAG_GRAFFITI)
+            .withBearerAuthSecurity()
+            .pathParam(PARAM_PUBKEY_TYPE)
+            .response(SC_OK, "Success response", RESPONSE_TYPE)
+            .withAuthenticationResponses()
+            .withNotFoundResponse()
+            .build());
+    this.keyManager = keyManager;
+  }
+
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    final BLSPublicKey publicKey = request.getPathParameter(PARAM_PUBKEY_TYPE);
+
+    final Optional<Validator> maybeValidator = keyManager.getValidatorByPublicKey(publicKey);
+    if (maybeValidator.isEmpty()) {
+      request.respondError(SC_NOT_FOUND, "Validator not found");
+      return;
+    }
+
+    String graffiti = maybeValidator.get().getGraffiti().map(this::processGraffitiBytes).orElse("");
+    request.respondOk(new GraffitiResponse(publicKey, graffiti));
+  }
+
+  private String processGraffitiBytes(final Bytes32 graffiti) {
+    return new String(graffiti.toArrayUnsafe(), StandardCharsets.UTF_8).strip().replace("\0", "");
+  }
+
+  static class GraffitiResponse {
+    private final Optional<String> publicKey;
+    private final String graffiti;
+
+    GraffitiResponse(final BLSPublicKey publicKey, final String graffiti) {
+      this.publicKey = Optional.of(publicKey.toHexString());
+      this.graffiti = graffiti;
+    }
+
+    Optional<String> getPublicKey() {
+      return publicKey;
+    }
+
+    String getGraffiti() {
+      return graffiti;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      GraffitiResponse that = (GraffitiResponse) o;
+      return Objects.equals(publicKey, that.publicKey) && Objects.equals(graffiti, that.graffiti);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(publicKey, graffiti);
+    }
```

Comment:
not sure we need these?

Reply:
Did for the testing, comparing the output